cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/main.go

```diff
@@ -31,3 +32,11 @@ func RegisterWithGRPCServer(s grpc.ServiceRegistrar) {
 func RegisterHTTPHandlerFromEndpoint(ctx context.Context, mux *runtime.ServeMux, endpoint string, opts []grpc.DialOption) error {
 	return v1alpha1.RegisterPackagesServiceHandlerFromEndpoint(ctx, mux, endpoint, opts)
 }
+
+// GetPluginDetail returns a core.plugins.Plugin describing itself.
+func GetPluginDetail() *plugins.Plugin {
+	return &plugins.Plugin{
+		Name:    "fluxv2.packages",
+		Version: "v1alpha1",
```

Comment:
What if a plugin exposes different versions?  Wiil we always load a very specific version or will let users invoke whichever version they want?

If the former, I'm ok with this Plugin signature, but, in case uses can use any version, perhaps `Version` should become `Versions: ["v1alpha1"]` instead, no?

Reply:
> What if a plugin exposes different versions? Will we always load a very specific version or will let users invoke whichever version they want?

The latter - we'll load whatever plugins are sitting in the plugin directory, so that could be `fluxv2.packages.v1alpha1` and `fluxv2.packages.v1`, and we'll serve both (not that people need to do that, but it enables supporting both).

> 
> If the former, I'm ok with this Plugin signature, but, in case uses can use any version, perhaps `Version` should become `Versions: ["v1alpha1"]` instead, no?

I did originally have it structured like this (a plugin with a list of versions), but changed to the current format because we're going to want to include the plugin with each package (or repo) that we return. As in, if a client requests packages generically, each package in the response will need to identify the plugin, I believe (I'm playing with this in the next PR).


> 
> Edit: ah, I see, it seems we are registering tuples `<name, version>`, so this `GetPluginDetail()` is returning what is expected. Perhaps it's a bit misleading to use "plugin" to refer to a combination of plugin_name+plugin_version, but I cannot think of a better option right now.

Well each version is a separate plugin and serving independently. You may choose to only ever have one version of fluxv2.packages available, but you may also choose to have two versions available, both independent.